tf2onnx/schemas.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""
+tf2onnx.schema
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import defaultdict, OrderedDict
+from onnx import defs
+
+
+ONNX_DOMAIN = ""
+
+
+class ONNXOpSchema(object):
+    """Wrapper for ONNX schema."""
+
+    def __init__(self, name, domain, since_version, attributes):
+        """Create a ONNX schema
+        Args:
+            name (str): op name
+            attributes (List[str]): valid attributes
+            domain (str): default value "" means it's ONNX domain
+            since_version (int): opset version, default is 1
+        """
+        self._name = name
+        self._domain = domain
+        self._attributes = attributes
+        self._since_version = since_version
+
+    @property
+    def attributes(self):
+        return self._attributes
+
+
+def map_schemas(onnx_schemas):
+    """Map schema in ONNX to tf2onnx.ONNXOpSchema"""
+    name_domain_version_schema_map = defaultdict(lambda: defaultdict(dict))
+    for s in onnx_schemas:
+        name = s.name
+        domain = s.domain
+        since_version = int(s.since_version)
+        name_domain_version_schema_map[name][domain][since_version] = \
+            ONNXOpSchema(name, domain, since_version, s.attributes)
+
+    ordered_version_map = defaultdict(lambda: defaultdict(OrderedDict))
```

Comment:
hmm, don't feel good about ```defaultdict``` here..
when someone uses schema mappings, lookup with a wrong key (either op name, domain, or version) is an error.
while using defaultdict might hide this error.